packages/pcd/semaphore-identity-v4/src/SemaphoreIdentityV4PCDPackage.ts

```diff
@@ -0,0 +1,116 @@
+import { DisplayOptions, PCDPackage, SerializedPCD } from "@pcd/pcd-types";
+import {
+  encodePrivateKey,
+  encodePublicKey,
+  podEdDSAPublicKeyHash
+} from "@pcd/pod";
+import { SemaphoreIdentityPCD } from "@pcd/semaphore-identity-pcd";
+import { requireDefinedParameter } from "@pcd/util";
+import { Identity } from "@semaphore-protocol/identity";
+import { derivePublicKey } from "@zk-kit/eddsa-poseidon";
+import { beBigIntToBuffer } from "@zk-kit/utils";
+import { sha256 } from "js-sha256";
+import JSONBig from "json-bigint";
+import { v4 as uuid } from "uuid";
+import {
+  SemaphoreIdentityV4PCD,
+  SemaphoreIdentityV4PCDArgs,
+  SemaphoreIdentityV4PCDClaim,
+  SemaphoreIdentityV4PCDProof,
+  SemaphoreIdentityV4PCDTypeName
+} from "./SemaphoreIdentityV4PCD";
+
+export function v4PublicKeyToCommitment(publicKey: string): string {
```

Comment:
>>if you intend to normalize on commitment being string everywhere
>that is what I intend

I think we're stuck with this being a decimal string for consistency with V3, and because string comparisons are used in a lot of places.  The micro-optimizer in me wishes it could be hex or base64 in order to be smaller.